### PR TITLE
Burst

### PR DESCRIPTION
**Pull request by andrewMacmurray**
+ removes square move
+ adds burst tile
+ begin to move Board / move / coord operations into modules
+ adds parcel as build system
+ convert JS to TS

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
   "scripts": {
     "deploy-production": "npm run build && firebase deploy -P production",
     "deploy-staging": "npm run build && firebase deploy -P staging",
-    "dev": "npm run css-build && parcel src/index.html",
-    "build": "npm run css-build && parcel build src/index.html",
+    "dev": "npm run css-build && elm-typescript-interop && parcel src/index.html",
```
Comment (nelsonic) [inline package.json]: @andrewMacmurray curious to read/hear your thoughts on this. 💭

File: package.json [hunk 2]
```diff
@@ -21,23 +21,24 @@
...
-    "tachyons-custom": "^4.5.4"
-  },
   "devDependencies": {
+    "@types/howler": "^2.1.1",
     "autoprefixer": "^6.7.3",
+    "bounce.js": "^0.8.2",
+    "elm": "^0.19.0",
     "elm-hot": "^1.1.2",
     "elm-live": "^3.0.6",
+    "elm-typescript-interop": "0.0.15",
```
Comment (nelsonic) [inline package.json]: is your plan to migrate all your `.js` to `.ts` ?

Comment (andrewMacmurray) [reply package.json]: I think there's only 1 `.js` file left which is config `postcss-config.js`.

File: package.json (hunk 1 above)
Comment (andrewMacmurray) [reply package.json]: Do you mean with elm-typescript-interop or using TypeScript more generally?

Comment (nelsonic) [reply package.json]: _Both_! 💡

Comment (andrewMacmurray) [reply package.json]: I used TypeScript at work on a previous project and it was a nice experience - it's very easy to gradually move existing JavaScript over (a valid JS file is a valid TS file), it adds basic sanity checks and has very nice IDE plugins (autocompletion / typechecking etc).

Using parcel for bundling also meant adding TypeScript was just changing the name of a file (I have a particular hatred of setting things up with Webpack so this was a big selling point for me).

With elm-typescript-interop there are some automatic sanity checks between the TS<->Elm boundary. Haven't had the need to write Decoders / Encoders / message routing logic for ports with this setup (although haven't put it through heavy use). We're using this on the current project at work and it's been nice.

My other thought with elm-typescript-interop is if there are ever any serious problems with it, it wouldn't be too much extra effort to maintain the `/Main/index.d.ts` file manually.

Definitely going off the "pain driven development" philosophy here and it's been pretty painless so far 😊

Comment (nelsonic) [reply package.json]: Insightful. thanks! 🥇

**Pull request merged**